src/libcore/iter/adapters/mod.rs

```diff
@@ -1609,6 +1609,100 @@ impl<I: Iterator, P> Iterator for TakeWhile<I, P>
     }
 }
 
+/// An iterator that only accepts elements while `predicate` returns `Some(_)`.
+///
+/// This `struct` is created by the [`take_while_map`] method on [`Iterator`]. See its
+/// documentation for more.
+///
+/// [`take_while_map`]: trait.Iterator.html#method.take_while
+/// [`Iterator`]: trait.Iterator.html
+#[must_use = "iterators are lazy and do nothing unless consumed"]
+#[unstable(feature = "iter_take_while_map", reason = "recently added", issue = "0")]
+#[derive(Clone)]
+pub struct TakeWhileMap<I, P> {
+    iter: I,
+    flag: bool,
+    predicate: P,
+}
+
+impl<I, P> TakeWhileMap<I, P> {
+    pub(super) fn new(iter: I, predicate: P) -> TakeWhileMap<I, P> {
+        TakeWhileMap { iter, flag: false, predicate }
+    }
+}
+
+#[unstable(feature = "iter_take_while_map", reason = "recently added", issue = "0")]
```

Comment:
Stability attributes currently don't work with trait impls

Reply:
Why then [there is] a `#[stable(feature = "rust1", since = "1.0.0")]` on `Iterator` impl for `TakeWhile`?

[there is]: https://github.com/rust-lang/rust/pull/66577/files/ebd6bb2d56d1e57edf42ff9a2ee5306e4312f836#diff-36a07733644b558bd8371f796386234fR1552

Reply:
It's unstability attributes that don't work. ;) There is no such thing as an unstable trait impl.

Reply:
I don't understand :( You said that there is no unstable trait impl, but
1) without `#[unsatble]` this code doesn't compile: ![image](https://user-images.githubusercontent.com/38225716/70691392-0ae56a00-1cca-11ea-8cb5-885e46b07361.png)
2) there are other `#[unstable]` impls, e.g.: [`impl TrustedLen for Rev`](https://github.com/rust-lang/rust/blob/ebd6bb2d56d1e57edf42ff9a2ee5306e4312f836/src/libcore/iter/adapters/mod.rs#L115-L117)
3) there are `#[stable]` impls too, e.g.: [`impl Iterator for Rev`](https://github.com/rust-lang/rust/blob/ebd6bb2d56d1e57edf42ff9a2ee5306e4312f836/src/libcore/iter/adapters/mod.rs#L39-L40)